### PR TITLE
Ruby: sinks for code injection via calls to `method`

### PR DESCRIPTION
**Pull request by p-**
Adds code injection sinks for calls to [`method`](https://ruby-doc.org/core-3.0.0/Method.html) with untrusted data.

E.g.
```ruby
m = method("exit")
m.call
```

(Note: the `public_method` and `singleton_method` are added for completeness, their usage in real-world projects is likely quite rare)

### CODE REVIEW

File: ruby/ql/test/library-tests/frameworks/core/Kernel.rb [hunk 1]
```diff
@@ -87,4 +87,8 @@ def run
   end
 end
 
-UnknownModule.system("ls")
\ No newline at end of file
+UnknownModule.system("ls")
+
+method("exit").call
+public_method("exit").call
+singleton_method("foo").call
```
Comment (hmac) [inline ruby/ql/test/library-tests/frameworks/core/Kernel.rb:94]: Does this also recognise calls with receivers, such as `Foo.method("exit")`?

File: ruby/ql/src/change-notes/2024-03-01-method-code-injection-sinks.md [hunk 2]
```diff
@@ -0,0 +1,4 @@
+---
+category: minorAnalysis
+---
+* Added code injection sinks for calls to `method` with untrusted data.
```
Comment (hmac) [inline ruby/ql/src/change-notes/2024-03-01-method-code-injection-sinks.md:4]:
```suggestion
* Calls to `Object#method`, `Object#public_method` and `Object#singleton_method` with untrusted data are now recognised as sinks for code injection.
```

**Review by hmac — COMMENTED** [review]
Some minor comments. As you say, the sink only really exists if there's a subsequent `.call` on the returned method object, but I think it's fine to skip that, as it is likely that the method is executed later on in the code anyway. We can always restrict this if it produces too many FPs.

File: ruby/ql/test/library-tests/frameworks/core/Kernel.rb (hunk 1 above)
Comment (p-) [reply ruby/ql/test/library-tests/frameworks/core/Kernel.rb:94]: Yes, good point 👍  I'll add more test cases for that

**Review by hmac — APPROVED** [review]

**Pull request merged**